builtin/logical/ssh/path_config_ca.go

```diff
@@ -36,17 +45,19 @@ For security reasons, the private key cannot be retrieved later.`,
 
 func (b *backend) pathCAWrite(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
 
-	publicKey := data.Get("public_key").(string)
-	privateKey := data.Get("private_key").(string)
-
-	_, err := ssh.ParsePrivateKey([]byte(privateKey))
-	if err != nil {
-		return nil, errutil.UserError{Err: fmt.Sprintf(`Unable to parse "private_key" as an SSH private key: %s`, err)}
-	}
+	var publicKey, privateKey string
+	var err error
+	if data.Get("generate_signing_key").(bool) {
```

Comment:
I think we should error if `generate_signing_key=true` and the user supplies public/private keys. Otherwise if they forget to add `generate_signing_key=false` vault will generate a key, but they might think vault is using the key they specified with little to no optics about which key vault is actually using.